app/lib/frontend/handlers/custom_api.dart

```diff
@@ -225,7 +229,13 @@ Future<shelf.Response> apiHistoryHandler(shelf.Request request) async {
 
 /// Handles requests for /api/search
 Future<shelf.Response> apiSearchHandler(shelf.Request request) async {
-  final searchQuery = parseFrontendSearchQuery(request.requestedUri, null);
+  SearchQuery searchQuery;
+  try {
+    searchQuery = parseFrontendSearchQuery(request.requestedUri, null);
+  } catch (e, st) {
```

Comment:
* Should we narrow to specific errors?
The catch-all might hide bugs in our parsing code.

* It seems there is a potential to have a `parseFrontendSearchQueryLoggingErrors` function.